### DISCUSSION TITLE
How to Show/Hide Fields in a Form Based on List of Results from a Query

### DISCUSSION THREAD

**Discussion by vaishnaray** [Budibase/budibase#11579] [Help]
HI,

I'm building a form that dynamically hides certain fields based on query results. Right now, I have the results of the query in a table and I want to show only the fields whose name matches the names in the table. 

When I use the Conditional UI feature, it only allows me to selected the SelectedRows in the table rather than all the rows in the table. I want to  see if the field name exists in the table, if it does not, then I want to hide it. 

As an example, I want to show this field in red because the name exists in the table above. Has anyone solved this before? Any help would be much appreciated!

<img width="573" alt="image" src="https://github.com/Budibase/budibase/assets/9454156/92b722b2-8492-43e6-b119-2f84822db7d4">

**Comment by ConorWebb96** [comment]
Hey @vaishnaray,

So in this case you only want `likelihood` to appear in the table when this is checked is that correct?

This will require filtering of the table's data provider. Hopefully you have the table and picker nested in 2 different data providers. Otherwise when you select the pickers option it will only show the one selected in the picker as well.

<img width="1462" alt="Screenshot 2023-08-23 at 10 06 29" src="https://github.com/Budibase/budibase/assets/126772285/96a7ed8e-5c17-4a89-b2db-70b159ac2335">
In the above screenshot this is how i've configured my components in the component tree. Additionally you will notice that there are 2 different data providers both are loading the same table. (You could do this with the custom option's source if you know the options, it will eliminate an extra data provider thus improving performance.)
<img width="100%" alt="Screenshot 2023-08-23 at 10 07 33" src="https://github.com/Budibase/budibase/assets/126772285/b1d874a3-1417-47fc-bb7f-ad6d4813e299">

The filter itself you will use your form thats the parent of both data providers and then select your options picker's value to be used as the binding that it checks against. (In my case im using the field gender to run this, in your case it would be `label_name`)
<img width="1153" alt="Screenshot 2023-08-23 at 11 30 59" src="https://github.com/Budibase/budibase/assets/126772285/b746c6bf-d1b8-4e4e-813e-ba190b797555">
<img width="1151" alt="Screenshot 2023-08-23 at 11 31 13" src="https://github.com/Budibase/budibase/assets/126772285/06460ac1-0872-44e6-8b19-7b368b33c2ff">


**Output**
![Screen Recording 2023-08-23 at 11 27 50](https://github.com/Budibase/budibase/assets/126772285/9405afa3-62ea-4da6-9c66-eff23f2b58a8)

I hope all of this makes sense and is helpful for you.

**Reply by vaishnaray** [reply]
Thank you! It's the other way. I want the likelihood field to show since likelihood is in the table and I want it hidden if it's not in the table. I'm not trying to filter the table, but rather show the options picker if it exists in the table.
